adapters/huaweiads/huaweiads.go

```diff
@@ -1192,6 +1281,24 @@ func (a *adapter) extractAdmPicture(content *content) (adm string, adWidth int64
 	return adm, adWidth, adHeight, nil
 }
 
+// for Interactiontype == appPromotion，clickUrl is intent
+func (a *adapter) getClickUrl(content *content) (clickUrl string, err error) {
+	if content.Interactiontype == appPromotion {
+		if content.MetaData.Intent != "" {
+			clickUrl = getDecodeValue(content.MetaData.Intent)
+		} else {
+			return "", errors.New("getClickUrl: content.MetaData.Intent is empty when interactiontype is appPromotion")
+		}
+	} else {
+		if content.MetaData.ClickUrl != "" {
+			clickUrl = content.MetaData.ClickUrl
+		} else if content.MetaData.Intent != "" {
+			clickUrl = getDecodeValue(content.MetaData.Intent)
+		}
```

Comment:
Can you add a test case covering this `else if` block?

Reply:
done